### ISSUE TITLE
Set training=False when loading pretrained models

### ISSUE THREAD

**Issue by elanmart** [pytorch/vision#386]
Hi, 

do you think it would be a good idea to call `model.eval()` when loading a pretrained network? 

I've spent a lot of time today trying to find out why my `inception` features were doing so poorly, and finally found out that the pretrained models are in `training` mode, and `dropout` kicks in making the features unsuable.

**Comment by alykhantejani** [comment]
hmm, not so sure about this as another common use for a pre-trained network is to continue to train it (finetune)

**Comment by lanpa** [comment]
How about
```python
def load_model(..., verbose=True):
  model = torch.load(...)
  if verbose:
    pass # print training/eval mode
  return model
```
?

**Comment by elanmart** [comment]
Oh, sorry for not following up on your response @alykhantejani . I wanted to say that I see your point, but perhaps we could have something like @lanpa proposed, or an explicit flag `mode='train'`?

**Comment by fmassa** [comment]
@elanmart how do you serialize your model, by saving the `state_dict` or the whole `nn.Module`?
I agree that it can be easy to miss that point, but except if we save the training state of each module inside the `state_dict`, I'm not sure how we could store that information.

Sure, we could try to propose a richer serialization format, but that might be trickier than it seems.

**Comment by alykhantejani** [comment]
@fmassa maybe a `mode` flag set to `train` by default in the load functions? this way we always set the mode of a loaded model (to either `train` (default) or `eval`) and it's clearer for the user from the signature that it will be in a mode?

**Comment by datumbox** [comment]
This issue appears to be stale, so I'm closing it. If you think it should remain open, feel free to reopen it.
